src/apdu_sign.c

```diff
@@ -635,6 +601,48 @@ void output_end(void) {
     }
 }
 
+// Called after all transaction outputs
+void outputs_end(void) {
+    // For some reason if we don't do this `G.dao_cell_owner` will get
+    // cleared?!?!?!
```

Comment:
Not "cleared"; G.dao_cell_owner is probably dead code in it's entirety, and was possibly reintroduced by a merge or something. It stays full but we don't even look at it for the prompts we still have for DAO.